### PR TITLE
1.0.1-beta1.1

### PR DESCRIPTION
**Pull request by timoj**
- Docs
- Code cleanup
- Resource ID's
- WebSocket endpoints CRUD
- Fixes for GUI clients
- Added background task run validation

### CODE REVIEW

File: lib/analyser/model.py [hunk 1]
```diff
@@ -38,15 +38,15 @@ async def add_model_from_dict(cls, dict_data):
         else:
             if await cls.get_model(model.model_name) is None:
                 cls.models.append(model)
-                await cls._update_cb(SUBSCRIPTION_CHANGE_TYPE_ADD, model.model_name, await EnodoModel.to_dict(model))
+                await cls._update_cb(SUBSCRIPTION_CHANGE_TYPE_ADD, await EnodoModel.to_dict(model))
```
Comment (joente) [inline lib/analyser/model.py:41]: Is there a very good reason to make the `to_dict` function async? It is currently just returning a `dict` without any async features, and I would actually not expect a "to_dict" function to be async.

File: lib/config/config.py [hunk 2]
```diff
@@ -146,6 +146,9 @@ def read_config(cls, path):
         :return:
         """
 
+        if not os.path.exists(path):
+            raise EnodoInvalidConfigException(f'Given config file does not exist or cannot be read')
```
Comment (joente) [inline lib/config/config.py:150]: Maybe include the `path` which is missing in the exception text? It is already a "formatting" string and might help to debug when this exception is raised

File: lib/analyser/model.py (hunk 1 above)
Comment (timoj) [reply lib/analyser/model.py:41]: It once called an other async func. This is indeed no longer the case, I will change it to a normal func

File: lib/config/config.py (hunk 2 above)
Comment (timoj) [reply lib/config/config.py:150]: Done

**Review by joente — APPROVED** [review]

**Pull request merged**